src/main/java/de/unistuttgart/overworldbackend/service/NPCService.java

```diff
@@ -17,49 +16,114 @@ public class NPCService {
   private NPCRepository npcRepository;
 
   @Autowired
-  private AreaService areaService;
+  private WorldService worldService;
+
+  @Autowired
+  private DungeonService dungeonService;
 
   @Autowired
   private NPCMapper npcMapper;
 
   /**
-   * Get a npc of an area
+   * Get a NPC from a world by its index and a lecture by its id
    *
-   * @throws ResponseStatusException if area or task with its id could not be found in the lecture
-   * @param lectureId the id of the lecture the npc is part of
-   * @param areaId the id of the area the npc is part of
-   * @param npcId the id of the npc searching for
-   * @return the found npc object
+   * @throws ResponseStatusException (404) if npc not found
+   * @param lectureId the id of the lecture
+   * @param worldIndex the index of the word
+   * @param npcIndex the index of the npc
+   * @return the found npc
    */
-  public NPC getNPCFromAreaOrThrowNotFound(final int lectureId, final UUID areaId, final UUID npcId) {
-    return areaService
-      .getAreaFromLectureOrThrowNotFound(lectureId, areaId)
+  public NPC getNPCFromWorld(final int lectureId, final int worldIndex, int npcIndex) {
+    return worldService
+      .getWorldByIndexFromLecture(lectureId, worldIndex)
       .getNpcs()
       .stream()
-      .filter(task -> task.getId().equals(npcId))
+      .filter(npc -> npc.getIndex() == npcIndex)
       .findAny()
       .orElseThrow(() ->
         new ResponseStatusException(
           HttpStatus.NOT_FOUND,
-          String.format("There is no npc with id %s world wit id %s in lecture with id %s.", npcId, areaId, lectureId)
+          String.format(
+            "There is no npc with index %s world with index %s in lecture with id %s.",
```

Comment:
```suggestion
            "There is no NPC %d in world %d, lecture %d.",
```
By the way, we probably have to fix this in multiple places.